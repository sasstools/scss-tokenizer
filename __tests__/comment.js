var scss = require('..');
var fs = require('fs');
var path = require('path');

var fixture = function(name) {
    return fs.readFileSync(
        path.join(__dirname, 'fixture', name)
    );
}

describe('Comment', function() {
    it('should tokenize a simple comment', function() {
        const tree = scss.tokenize(fixture('simple-comment.scss'));
        expect(tree).toMatchSnapshot();
    });

    it('should tokenize a multiline comment', function() {
        const tree = scss.tokenize(fixture('multiline-comment.scss'));
        expect(tree).toMatchSnapshot();
    });

    it('should tokenize a docblock comment', function() {
        const tree = scss.tokenize(fixture('docblock-comment.scss'));
        expect(tree).toMatchSnapshot();
    });

    it('should tokenize a comment that does not have a space before the end', function() {
        const tree = scss.tokenize(fixture('comment-with-trailing-space.scss'));
        expect(tree).toMatchSnapshot();
    });
});
