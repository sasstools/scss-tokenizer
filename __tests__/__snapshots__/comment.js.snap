// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment should tokenize a comment that does not have a space before the end 1`] = `
Array [
  Array [
    "startComment",
    "/*",
    1,
    2,
  ],
  Array [
    "word",
    "my",
    1,
    3,
    1,
    4,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "word",
    "comment",
    1,
    6,
    1,
    12,
  ],
  Array [
    "endComment",
    "*/",
    1,
    14,
  ],
]
`;

exports[`Comment should tokenize a docblock comment 1`] = `
Array [
  Array [
    "startComment",
    "/*",
    1,
    2,
  ],
  Array [
    "word",
    "*",
    1,
    3,
    1,
    3,
  ],
  Array [
    "newline",
    "
",
    2,
    0,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "word",
    "*",
    2,
    2,
    2,
    2,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "word",
    "line",
    2,
    4,
    2,
    7,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "word",
    "1",
    2,
    9,
    2,
    9,
  ],
  Array [
    "newline",
    "
",
    3,
    0,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "word",
    "*",
    3,
    2,
    3,
    2,
  ],
  Array [
    "newline",
    "
",
    4,
    0,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "word",
    "*",
    4,
    2,
    4,
    2,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "word",
    "line",
    4,
    4,
    4,
    7,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "word",
    "2",
    4,
    9,
    4,
    9,
  ],
  Array [
    "newline",
    "
",
    5,
    0,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "endComment",
    "*/",
    5,
    3,
  ],
]
`;

exports[`Comment should tokenize a multiline comment 1`] = `
Array [
  Array [
    "startComment",
    "/*",
    1,
    2,
  ],
  Array [
    "newline",
    "
",
    2,
    0,
  ],
  Array [
    "word",
    "my",
    2,
    1,
    2,
    2,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "word",
    "comment",
    2,
    4,
    2,
    10,
  ],
  Array [
    "newline",
    "
",
    3,
    0,
  ],
  Array [
    "endComment",
    "*/",
    3,
    2,
  ],
]
`;

exports[`Comment should tokenize a simple comment 1`] = `
Array [
  Array [
    "startComment",
    "/*",
    1,
    2,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "word",
    "my",
    1,
    4,
    1,
    5,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "word",
    "comment",
    1,
    7,
    1,
    13,
  ],
  Array [
    "space",
    " ",
  ],
  Array [
    "endComment",
    "*/",
    1,
    16,
  ],
]
`;
